src/librustc/hir/print.rs

```diff
@@ -1314,7 +1314,12 @@ impl<'a> State<'a> {
                 }
                 self.bclose_(expr.span, indent_unit)?;
             }
-            hir::ExprClosure(capture_clause, ref decl, body, _fn_decl_span) => {
+            hir::ExprClosure(capture_clause, ref decl, body, _fn_decl_span, gen) => {
+                if gen.is_some() {
+                    self.head("gen")?;
+                    space(&mut self.s)?;
+                }
```

Comment:
Immovable types is a separate orthogonal proposal. Immovable generators would be a extension to generators which depend on immovable types. Immovable generators are very important for async I/O experiments however.

I do intend to implement immovable generators, but landing that in master would block on the immovable types RFC.